src/datapump/sync/sync.py

```diff
@@ -533,65 +537,117 @@ def get_raster_job(
         return raster_job
 
     def get_latest_release(self) -> Tuple[str, List[str]]:
-        """
-        Get the version of the latest *complete* release in GCS
-        """
+        # UMD's GLAD release scheme goes something like this:
+        # GLAD alerts are released as a set of tiles that contain all the
+        # alerts for a given year, placed in
+        # GLADalert/C2/{year_of_release}/{month_day_of_release}
+        # They start out as provisional, which means pixel values may be
+        # modified following a QA process (with the new values appearing
+        # in subsequent releases). At some point (typically around July)
+        # there will be some final QA, after which the alerts for the
+        # previous year will be put in the GLADalert/C2/{year_of_data}/final
+        # folder.
+        # There will be some time during which the processing of current
+        # year alerts overlaps the processing of the previous year's alerts,
+        # but the tiles have the year of the data in the file names to
+        # differentiate them (there are two bands/sets of tiles, named
+        # like alert{two_digit_year} and alertDate{two_digit_year}).
+        # So.
+        # For a given year's data, first see if it's already been finalized
+        # (i.e. there are tiles in year/final). If not, find the last day it
+        # was released as provisional data (which may be today or some time
+        # in the past: As of this writing they have stopped including the 2021
+        # data in daily updates but have not yet put anything in the "final"
+        # folder).
+        today = self.get_today()
+        month_day = today.strftime("%m_%d")
+
+        source_uris: List[str] = []
+        release_version: str = "v" + today.strftime("%Y%m%d")
+
+        for target_year in range(self.start_year, today.year + 1):
```

Comment:
Just to make sure I understand the logic here - for every year, we check for a final folder, and if it doesn't exist, e check the last date? E.g. for this year, we will (eventually) find a final folder for 2021, but we won't for 2022 until next July, so we it'll just keep using the latest date for the 2022 alerts?

Reply:
Yep!